src/chapters/http/how-the-internet-works.rst

```diff
@@ -0,0 +1,247 @@
+   :alt: A server surrounded by several client computers, connected by arrows.
+   :height: 400px
+
+   Several clients communicating with a server.
+
+.. admonition:: Fun Fact
+
+   In the client-server model, the server may sometimes be on the *same* computer as the client. This is often the case when a programmer is building a web application. The in-progress, development version of the application is on their laptop, as is their browser that they use to test the app.
+
+Protocols
+---------
+
+.. index::
+   single: url; protocol
+
+A **protocol** is a standard for communication between computers. Most web communication uses *three* protocols, in fact.
+
+.. list-table:: Common Web Protocols
+   :header-rows: 1
+
+   * - Protocol
+     - Full Name
+     - Role
+   * - HTTP
+     - Hypertext Transfer Protocol
+     - High-level web communication for transferring files and information, including:
+
+       - Code files that create web sites (HTML, CSS, C Sharp, etc.)
+       - images and other media
+       - form submissions
+
+   * - TCP/IP
+     - Transmission Control Protocol / Internet Protocol
+     - Low-level web communication for transferring small chunks of raw data known as *packets*
+   * - DNS
+     - Domain Name Service
+     - Translates human-friendly names into server addresses
+
+A thorough understanding of each of these protocols is well beyond the scope of this class. However, as a web developer it is important you have a general understanding of their roles. Each protocol has a different and critical job in enabling web communication.
+
+HTTP
+^^^^
+
+.. index:: ! HTTP, ! HTTPS
+
+**HTTP** is the most important protocol for web developers to understand, which you may have guessed from the title of this chapter. It specifies how requests for common web data---such as HTML files or images---should be structured, as well as responses to such requests. The details of request and response message structure are the topic of the rest of this chapter. 
+
+**HTTPS** refers to the HTTP protocol used with a secure connection. A secure connection encrypts so that it can't be read while in-transit. The data is encrypted by the server/client before being transmitted, and decrypted once it is received by the client/server. The precise details of how such encryption works is beyond the scope of this course. 
+
+TCP/IP
+^^^^^^
+
+.. index:: ! TCP/IP
+
+**TCP/IP** is a low-level protocol that is quite complicated. For our purposes, it is important only to know that TCP/IP is the standard that allows *raw data* to get from one place to another on the Internet. 
+
+When a server sends a file back to a client, that file must physically be sent across a series of network components, including cables, routers, and switches. Files are broken down into *packets*---small chunks of a standard size---that are individually sent from one location to the next, until arriving at their final destination and being reassembled.
```

Comment:
I do like how this is explained, but wonder if its too simplified?  I looked at this article https://www.techopedia.com/definition/2460/transmission-control-protocolinternet-protocol-tcpip and this video https://www.youtube.com/watch?v=614QGgw_FA4 to better understand.  Maybe not change anything but add in a link to something that gives more details for those who want a bit more understanding?